### PR TITLE
Improve Cupertino docs

### PR DESCRIPTION
**Pull request by guidezpl**
## Description

Adds "See also" section to cupertino library
Adds 3 snippets for `CupertinoApp` (2 with screenshots require https://github.com/flutter/assets-for-api-docs/pull/129 to merge first)
Adds a dartpad for `CupertinoNavigationBar`
Adds a dartpad for `CupertinoPageScaffold`

## Related Issues

https://github.com/flutter/flutter/issues/72926

## Tests

No tests, only documentation.

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/app.dart [hunk 1]
```diff
@@ -46,6 +46,75 @@ import 'theme.dart';
...
+///       );
+///     }
+///   },
+/// )
+/// ```
+/// {@end-tool}
+///
+/// {@tool snippet}
+/// This example shows how to create a [CupertinoApp] that defines a [theme] that
+/// will be used for cupertino widgets in the app.
```
Comment (shihaohong) [inline packages/flutter/lib/src/cupertino/app.dart:98]: nit:
```suggestion
/// will be used for Cupertino widgets in the app.
```

File: packages/flutter/lib/src/cupertino/nav_bar.dart [hunk 2]
```diff
@@ -197,6 +197,33 @@ bool _isTransitionable(BuildContext context) {
...
+///       middle: const Text('Sample Code'),
+///     ),
+///     child: Column(
+///         children: [
+///           Container(height: 50, color: CupertinoColors.systemRed),
+///           Container(height: 50, color: CupertinoColors.systemGreen),
+///           Container(height: 50, color: CupertinoColors.systemBlue),
+///           Container(height: 50, color: CupertinoColors.systemYellow),
+///         ]
+///     ),
```
Comment (shihaohong) [inline packages/flutter/lib/src/cupertino/nav_bar.dart:221]: nits:
```suggestion
///     child: Column(
///       children: [
///         Container(height: 50, color: CupertinoColors.systemRed),
///         Container(height: 50, color: CupertinoColors.systemGreen),
///         Container(height: 50, color: CupertinoColors.systemBlue),
///         Container(height: 50, color: CupertinoColors.systemYellow),
///       ],
///     ),
```

File: packages/flutter/lib/src/cupertino/page_scaffold.dart [hunk 3]
```diff
@@ -18,6 +18,37 @@ import 'theme.dart';
 /// encloses the [ScrollView]. The [ScrollView.primary] flag is used to connect
 /// a [ScrollView] to the enclosing [PrimaryScrollController].
 ///
+/// {@tool dartpad --template=stateful_widget_cupertino}
+/// This example shows a [CupertinoPageScaffold] with a [ListView] as a [child].
+/// The [CupertinoButton] is connected to a callback that increments a counter.
+///
```
Comment (shihaohong) [inline packages/flutter/lib/src/cupertino/page_scaffold.dart:24]: nit: Just one new line is enough here
```suggestion
```

File: packages/flutter/lib/src/cupertino/page_scaffold.dart [hunk 4]
```diff
@@ -18,6 +18,37 @@ import 'theme.dart';
...
+///
+/// ```dart
+/// int _count = 0;
+///
+/// Widget build(BuildContext context) {
+///   return CupertinoPageScaffold(
+///     // Uncomment to change the background color
+///     // backgroundColor: CupertinoColors.systemPink,
```
Comment (shihaohong) [inline packages/flutter/lib/src/cupertino/page_scaffold.dart:32]: This comment about the background color seems unrelated to the sample itself. Is this intended? If it is, add it to the dartpad description above

**Review by shihaohong — APPROVED** [review]
LGTM with nits and once the `assets-for-api-docs` change is merged.

File: packages/flutter/lib/src/cupertino/page_scaffold.dart (hunk 4 above)
Comment (guidezpl) [reply packages/flutter/lib/src/cupertino/page_scaffold.dart:32]: Yes, modified

**Pull request merged**